packages/xstate-graph/src/graph.ts

```diff
@@ -381,3 +382,66 @@ export function toDirectedGraph(stateNode: StateNode): DirectedGraphNode {
 
   return graph;
 }
+
+export function getPathFromEvents<
+  TContext = DefaultContext,
+  TEvent extends EventObject = EventObject
+>(
+  machine: StateMachine<TContext, any, TEvent>,
+  events: Array<TEvent>
+): StatePath<TContext, TEvent> {
+  const optionsWithDefaults = getValueAdjMapOptions<TContext, TEvent>({
+    events: events.reduce((events, event) => {
+      events[event.type] ??= [];
+      events[event.type].push(event);
+      return events;
+    }, {})
```

Comment:
This doesn't do anything about duplicates, figured it doesn't matter. But wanted to highlight.